File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    let answer = 0;
    function targetNumber(array, index, value, target) {
        if (index === array.length) {
            if (target === value) answer++;
            return;
        }

        targetNumber(array, index + 1, (value + array[index]), target);
        targetNumber(array, index + 1, (value - array[index]), target);
    }

    targetNumber(numbers, 0, 0, target);
    return answer;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = [0]

    def dfs(value, index):
        if index == len(numbers):
            if value == target:
                answer[0] += 1
            return

        dfs(value + numbers[index], index + 1)
        dfs(value - numbers[index], index + 1)

    dfs(0, 0)

    return answer[0]